app.ts

```diff
@@ -0,0 +1,205 @@
+			this.firstRecord = this.firstRecord < 0 ? 0 : this.firstRecord;
+			this.firstRecord = this.firstRecord > RECORD_COUNT - 1 ? RECORD_COUNT - 1 : this.firstRecord;
+			alert('please select a number between 0 and ' + (RECORD_COUNT - 1));
+			return;
+		}
+
+		this.myTable.innerHTML = ''; // empty table div to fill in a new table
+
+		// get table height based on  remaining window size after buttons
+		CONTENT_HEIGHT = (window.innerHeight - CONTROLS_HEIGHT);
+		this.myTable.setAttribute("height", "CONTENT_HEIGHT");
+
+		// calculate row number based on table header + row heights + borders in CSS and ensure its never less than 1
+		let numRows: number = Math.floor((CONTENT_HEIGHT - CONTROLS_HEIGHT - 26) / 46);
+		numRows = numRows <= 0 ? 1 : numRows;
+
+		// set number of records to get from server and ensure that the last record doesnt exceed the limit.
+		this.endRecord = this.firstRecord - 1 + numRows;
+		this.endRecord = (this.endRecord >= RECORD_COUNT ? RECORD_COUNT - 1 : this.endRecord);
+
+		// write html for table header
+		for (const i of columnsArray) {
+			const th = document.createElement('th');
+			this.myTable.appendChild(th);
+
+			th.appendChild(document.createTextNode(" " + i));
+		}
+		getTable(this.firstRecord, this.endRecord, this);
+	}
+
+	// function used by the next page buttton, requires the last entry from previously generated table
+	nextPage() {
+
+		this.firstRecord = this.endRecord + 1;
+		this.firstRecord = this.firstRecord > RECORD_COUNT - 1 ? RECORD_COUNT - 1 : this.firstRecord;
+		// getTable(this.firstRecord, this.endRecord, this);
+		this.generateTable();
+	}
+
+	firstPage() {
+		this.firstRecord = 0;
+		// getTable(this.firstRecord, this.endRecord, this);
+		this.generateTable();
+	}
+
+	// function used by the next page buttton, requires the first and last entry from previously generated table
+	prevPage() {
+		// if the new first entry is less than zero, take user to the first page 
+		this.firstRecord = this.firstRecord - (this.endRecord - this.firstRecord) - 1;
+		this.firstRecord = this.firstRecord < 0 ? 0 : this.firstRecord;
+		// getTable(this.firstRecord, this.endRecord, this);
+		this.generateTable();
+	}
+}
+
+// hint for the user
+function helpFunc() {
+	alert(
+		"Table for viewing records with IDs 0 to" + (RECORD_COUNT - 1) + " (Numbers Only)."
+		+ "\nTo start from a specific record enter a record ID and click 'Go To'.");
+}
+
+let customPage = () => {
+	let inputVal = parseInt(this.recordNum.value);
+	if (isNaN(inputVal)) {
+		alert('Please type in a number')
+		return;
+	}
+	tableElement.firstRecord = inputVal;
+	tableElement.generateTable();
+}
+
+window.onresize = () => {
+	const doneResizing = () => {
+		tableElement.generateTable();
+	}
+
+	clearTimeout(resizeTimer);
+	resizeTimer = setTimeout(doneResizing, 500);
+}
+
+let contentsDiv2: HTMLElement;
+let tableElement: tableGenerator;
+
+// load page for the first time and create buttons
+window.onload = () => {
```

Comment:
I would name this function something like `generateTable` and then go `window.onload = generateTable' then you know what the code is going to do without having to read it first.